### PR TITLE
Hide watermark label when clicked on it.

### PR DESCRIPTION
**Pull request by jbialobr**
Fixes https://github.com/gitextensions/gitextensions/issues/4683#issuecomment-375118791

Changes proposed in this pull request:
 - Hide watermark label when clicked on it.
 - Focus FilterComboBox.
 
What did I do to test the code and ensure quality:
 - Tested manually.

Has been tested on (remove any that don't apply):
 - Windows 10

### CODE REVIEW

File: GitUI/UserControls/FileStatusList.cs [hunk 1]
```diff
@@ -1307,6 +1307,12 @@ private void FilterComboBox_LostFocus(object sender, EventArgs e)
         private Regex _filter;
 
         #endregion Filtering
```
Comment (gerhardol) [inline GitUI/UserControls/FileStatusList.cs]: Nitty picky: The method should be in the region

File: GitUI/UserControls/FileStatusList.cs [hunk 2]
```diff
@@ -1307,6 +1307,12 @@ private void FilterComboBox_LostFocus(object sender, EventArgs e)
...
+
+        private void FilterWatermarkLabel_Click(object sender, EventArgs e)
+        {
+            FilterWatermarkLabel.Visible = false;
```
Comment (gerhardol) [inline GitUI/UserControls/FileStatusList.cs]: Nitty picky: Statement not needed,done in Focus()

File: GitUI/UserControls/FileStatusList.cs (hunk 1 above)
Comment (jbialobr) [reply GitUI/UserControls/FileStatusList.cs]: Thanks!

**Comment by jbialobr** [comment]
>Longer term it seems like this should be bundled into a WatermarkComboBox control.

More generally: whenever you think of using a `region` consider separating the thing into its own class.

**Pull request merged**